pkgs/servers/ldap/lldap/rustc-wasm.nix

```diff
@@ -0,0 +1,17 @@
+{ rustc , stdenv }:
+
+# https://github.com/NixOS/nixpkgs/issues/89426
+(rustc.override {
+  stdenv = stdenv.override {
+    targetPlatform = stdenv.targetPlatform // {
+      parsed = {
+        cpu.name = "wasm32";
+        vendor.name = "unknown";
+        kernel.name = "unknown";
+        abi.name = "unknown";
+      };
+    };
+  };
+}).overrideAttrs (attrs: {
+  configureFlags = attrs.configureFlags ++ ["--set=build.docs=false"];
+})
```

Comment:
Good move extracting this. Though I'm wondering if it wouldn't make sense pulling it into `all-packages.nix` and (at least partially) resolving https://github.com/NixOS/nixpkgs/issues/89426 ...

Reply:
Sounds like a job for a dedicated and very thought out PR, one either expanding `pkgsCross.wasi32.rustc` to cover `wasm32-unknown-unknown`, or one that introduces `pkgsCross.wasm32`

Reply:
In the short term I still think it's better to expose the rustc-wasm derivation as it's own attribute in all-packges than doing this relative path call package reach

Reply:
One hand, yes. That would be neat and the main reason I didn't do it because I know little about cross compiling and less about wasm/wasi.

Other hand, if this addresses an immediate need of more than one package, I'd think it had a place in all-packages, even if it's no 10/10 solution. Also: From the (very limited) exposure I had, the wasm ecosystem (so far) doesn't seem to rely on binary libraries (`.a` or `.so`), and most projects just seem to compile a bunch of rust source in one go. So the question stands which wasm projects could even profit from the power of a full blown cross nixpkgs at this point in time.

Reply:
consider me convinced. The most recent force push moves it to `rustc-wasm32`